WalletWasabi.Fluent/Helpers/TransactionFeeHelper.cs

```diff
@@ -117,27 +116,42 @@ public static TimeSpan CalculateConfirmationTime(double targetBlock)
 		return time;
 	}
 
-	public static bool TryGetMaximumPossibleFeeRate(decimal percentageOfOverpayment, Wallet wallet, FeeRate currentFeeRate, out FeeRate maximumPossibleFeeRate)
+	public static async Task<bool> TrySetMaxFeeRateAsync(Wallet wallet, TransactionInfo info)
 	{
-		maximumPossibleFeeRate = FeeRate.Zero;
+		var newInfo = info.Clone();
+		var lastWrongFeeRate = new FeeRate(0m);
+		var lastCorrectFeeRate = new FeeRate(0m);
 
-		if (percentageOfOverpayment <= 0 || !TryGetFeeEstimates(wallet, out var feeEstimates))
+		do
 		{
-			return false;
-		}
-
-		var maxPossibleFeeRateInSatoshiPerByte = (currentFeeRate.SatoshiPerByte / percentageOfOverpayment) * 100;
-		maximumPossibleFeeRate = new FeeRate(maxPossibleFeeRateInSatoshiPerByte);
-
-		var feeChartViewModel = new FeeChartViewModel();
-		feeChartViewModel.UpdateFeeEstimates(feeEstimates.Estimations);
+			try
+			{
+				await Task.Run(() => TransactionHelpers.BuildTransaction(wallet, newInfo, tryToSign: false));
+				lastCorrectFeeRate = newInfo.FeeRate;
+				var increaseBy = lastWrongFeeRate.SatoshiPerByte == 0 ? newInfo.FeeRate.SatoshiPerByte : (lastWrongFeeRate.SatoshiPerByte - newInfo.FeeRate.SatoshiPerByte) / 2;
```

Comment:
In the first iteration if `newInfo.FeeRate.SatoshiPerByte` is zero it will never increase and the loop will go forever.

Reply:
fixed